app/controllers/custom_fields_controller.rb

```diff
@@ -24,20 +34,69 @@ def create
     end
   end
 
+  def update
+    respond_to do |format|
+      format.json do
+        @custom_field.update_attributes(custom_field_params)
+        if @custom_field.save
+          render json: { status: :ok }
+        else
+          render json: @custom_field.errors.to_json,
+                 status: :unprocessable_entity
+        end
+      end
+    end
+  end
+
+  def destroy_html
+    respond_to do |format|
+      format.json do
+        render json: {
+          html: render_to_string(
+            partial: 'samples/delete_custom_field_modal_body.html.erb'
+          )
+        }
+      end
+    end
+  end
+
+  def destroy
+    respond_to do |format|
+      format.json do
+        if @custom_field.destroy
+          render json: { status: :ok }
+        else
+          render json: { status: :unprocessable_entity }
+        end
+      end
+    end
+  end
+
   private
 
+  def load_vars
+    @custom_field = CustomField.find_by_id(params[:id])
+    @custom_field = CustomField.find_by_id(
+      params[:custom_field_id]
+    ) unless @custom_field
+    render_404 unless @custom_field
+  end
```

Comment:
Maybe the code should be changed so that only one key is used for getting the custom field id. I assume in some places 'id' is used and in other 'custom_field_id' is used, so I suggest that everywhere only 'custom_field_id' is used.